### PR TITLE
Bug 1486612 - Convert mozilla-services/mozphab to use Circle CI 2.0

### PR DESCRIPTION
**Pull request by dklawren**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -32,7 +35,6 @@ RUN apk --no-cache --update add \
     libpng \
     make \
     mariadb-client \
-    mariadb-client-libs \
```
Comment (ckolos) [inline Dockerfile]: No longer required?

Comment (dklawren) [reply Dockerfile]: Yeah it has been removed in more recent versions of Alpine but mariadb-client is still there. I do not see any errors while building so I assumed it was working ok.

File: .circleci/config.yml [hunk 2]
```diff
@@ -40,12 +40,23 @@ jobs:
       - run: invoke build
       - run: invoke imageid | tee /artifacts/docker-image-shasum256.txt
       - deploy:
+          name: dockerhub latest
+          branch: master
           command: |
-            if [[ !-z $DOCKERHUB_REPO ]]; then
-              docker login -e "$DOCKER_EMAIL" -u "$DOCKER_USER" -p "$DOCKER_PASS"
+            if [[ ! -z $DOCKERHUB_REPO ]]; then
```
Comment (ckolos) [inline .circleci/config.yml]: "$DOCKERHUB_REPO" not $DOCKERHUB_REPO for string comparison

Comment (ckolos) [reply .circleci/config.yml]: also be consistent in use of [[]] vs. []; [[]] not needed here

File: .circleci/config.yml [hunk 3]
```diff
@@ -40,12 +40,23 @@ jobs:
...
+              docker login "$DOCKER_EMAIL" -u "$DOCKER_USER" -p "$DOCKER_PASS"
               docker tag  "${DOCKERHUB_REPO}" "${DOCKERHUB_REPO}:${CIRCLE_SHA1}"
               docker push "${DOCKERHUB_REPO}:${CIRCLE_SHA1}"
               docker push "${DOCKERHUB_REPO}:latest"
             fi
+      - deploy:
+          name: dockerhub tags
+          tags: /.*/
+          command: |
+            if [ ! -z $DOCKERHUB_REPO ]; then
```
Comment (ckolos) [inline .circleci/config.yml]: Ditto from above

**Comment by dklawren** [comment]
Ready for normal review.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          name: Create version.json
+          command: |
+            invoke version > version.json && \
+            mkdir /artifacts && \
+            cp version.json /artifacts/version.json
+      - run: invoke build
+      - run: invoke imageid | tee /artifacts/docker-image-shasum256.txt
+      - deploy:
+          command: |
+            [[ -n "$DOCKERHUB_REPO" && -n "$DOCKER_USER" && -n "$DOCKER_PASS" ]] || exit 0
```
Comment (ckolos) [inline .circleci/config.yml]: shouldn't that be exit 1 if they're not set?

**Pull request merged**